conclave-common/src/main/kotlin/com/r3/conclave/common/internal/CallInterface.kt

```diff
@@ -0,0 +1,49 @@
+package com.r3.conclave.common.internal
+
+import java.nio.ByteBuffer
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * This class serves as the basis for enclave and host call interfaces.
+ * It both services incoming calls and handles outgoing calls.
+ * It maintains an internal mapping of incoming call types to registered call handlers.
+ */
+abstract class CallInterface<OUTGOING_CALL_TYPE, INCOMING_CALL_TYPE> {
+    companion object {
+        private val EMPTY_BYTE_BUFFER: ByteBuffer = ByteBuffer.wrap(ByteArray(0)).asReadOnlyBuffer()
+    }
+
+    private val callHandlers = ConcurrentHashMap<INCOMING_CALL_TYPE, CallHandler>()
+
+    /**
+     * Execute a call and maybe get a return buffer.
+     */
+    abstract fun initiateOutgoingCall(callType: OUTGOING_CALL_TYPE, parameterBuffer: ByteBuffer = EMPTY_BYTE_BUFFER): ByteBuffer?
```

Comment:
What does "initiate" mean? It implies the execution can be asynchronous, but I presume that's not the case. As you've documented, "execute" might be clearer.